libs/area_threading.py

```diff
@@ -33,6 +34,16 @@ def __init__(self, config, area):
 
     def run(self):
         self.engine = AreaEngine(self.config, self.area)
+        while True:
+            try:
+                self.engine.process_area()
+            except Exception as e:
+                logging.error(e, exc_info=True)
+                # Sleep the thread for 5 seconds and try to process the area again
+                time.sleep(5)
+                logging.info(f"Exception processing area {self.area['name']}")
+                logging.info("Restarting the area processing")
+
         self.engine.process_area()
```

Comment:
Should this be deleted?

Reply:
Maybe we can delete the `time.sleep(5)`. 
I put it because when the `Exception` is fatal, you get the log flooded by the Exception trace. However, the `sleep` doesn't fix that issue. I will change it implementing a max retry logic.

Reply:
I added the `MaxThreadRestarts` parameter in the config file.
I also removed the line `self.engine.process_area() `.